app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedFragment.java

```diff
@@ -0,0 +1,216 @@
+            presenter.setShowFavorite(showFavoriteNews);
+            presenter.setCategory(category);
+        }
+
+    }
+
+    private void initRecyclerView() {
+        adapter = new NewsFeedAdapter(context);
+        recyclerView.setLayoutManager(new LinearLayoutManager(context));
+        recyclerView.setAdapter(adapter);
+        adapter.setOnItemClickListener(new NewsFeedAdapter.OnNewsItemClickListener() {
+            @Override
+            public void onChangeFavoriteStateClick(int position) {
+                presenter.changeNewsFavoriteState(position);
+            }
+
+            @Override
+            public void OnRecyclerItemClick(int position) {
+                presenter.openNewsDetails(position);
+            }
+        });
+    }
+
+    @Override
+    public void setFavoriteIcon(Boolean showOnlyFavorite) {
+        if (showOnlyFavorite) {
+            favoriteNewsButton.setImageDrawable(IconicStorage.getYellowStarBorderless(context));
+        } else {
+            favoriteNewsButton.setImageDrawable(IconicStorage.getWhiteStarBorderless(context));
+        }
+    }
+
+    @Override
+    public void setCategoryTitle(Category category) {
+        updateCategoryNameOnDescription(Category.getDisplayName(category));
+    }
+
+    @Override
+    public void updateNewsItem(int position) {
+        adapter.notifyItemChanged(position);
+    }
+
+    @Override
+    public void removeNewsItem(int position) {
+        adapter.notifyItemRemoved(position);
+    }
+
+    @Override
+    public void showProgressBar() {
+        progressBar.setVisibility(ProgressBar.VISIBLE);
+    }
+
+    @Override
+    public void hideProgressBar() {
+        progressBar.setVisibility(ProgressBar.GONE);
+
+    }
+
+    private void setEmptyViewOnDisplay() {
+        recyclerView.setVisibility(View.GONE);
+        emptyView.setVisibility(View.VISIBLE);
+    }
+
+    private void setRecyclerViewOnDisplay() {
+        recyclerView.setVisibility(View.VISIBLE);
+        emptyView.setVisibility(View.GONE);
+    }
+
+    @Override
+    public void updateNewsListUI(List<NewsItem> news) {
+        Log.d(TAG, "updateNewsListUI: ");
+        if (news == null || news.isEmpty()) {
+            setEmptyViewOnDisplay();
+        } else {
+            setRecyclerViewOnDisplay();
+            showNews(news);
+        }
+    }
+
+    private void showNews(List<NewsItem> news) {
+        Log.d(TAG, "showNews: " + news);
+        setRecyclerViewOnDisplay();
+        adapter.setItems(news);
+        adapter.notifyDataSetChanged();
+    }
+
+    private void updateCategoryNameOnDescription(String category) {
+        Log.d(TAG, "updateCategoryNameOnDescription: ");
+        descriptionView.setText(category);
+    }
+
+    @Override
+    public void showNewsDetailsUI(String url) {
+        Navigator.openWebView(url, context);
+    }
+
+    @Override
+    public void onCategorySelected(Category selectCategory) {
+        presenter.setCategory(selectCategory);
```

Comment:
This should call a `presenter.onCategorySelected` method which will call required `View` methods and process `selectedCategory`.